src/routes/solid-router/reference/data-apis/cache.mdx

```diff
@@ -24,7 +20,35 @@ This cache accomplishes the following:
    Any user based navigation or link click bypasses it.
    Upon revalidation or new fetch the cache is updated.
 
-Using it with a load function:
+### Arguments
+
+| argument | type | description |
+| -------- | ---- | ----------- |
+| fn | `(...args: any) => any` | A function whose return value you'd like to be cached. This function can have any arguments you'd like, and the arguments will be deterministically serialized and used as part of the unique key for the cache. This serialization process uses `JSON.stringify`, but first sorts the key/value pairs of objects to ensure `{ name: 'Ryan', awesome: true }` serializes to the same value as `{ awesome: true, name: 'Ryan' }`. |
+| name | string | Any arbitrary string that you'd like to use as the rest of the cache key. Because the cache used internally by `cache` is shared by all cached functions, this string should be unique for each unique function passed to `cache`. If you instead use the same key for two different cached functions, it's possible that one will give you back the return value from the other. |
+
+### Return Value
+
+The return value is a `CachedFunction`, a function that has the same signature as the function you passed to `cache`.
+This cached function stores the return value (using the cache key as described above) and temporarily avoids re-running the passed function again for the same set of arguments under most circumstances, even if you call the created function multiple times.
+The returned function also has `key` and `keyFor` properties, described below.
```

Comment:
This example will break suggestions but can you include this here please.

```suggestion
### `.key` and `.keyFor`

Cached functions provide `.key` and `.keyFor`, which are useful when retrieving the keys used in cases involving invalidation:

```ts
let id = 5;
getUser.key; // returns "users"
getUser.keyFor(id); // returns "users[5]"
```

```